### ISSUE TITLE
csi-rbdplugin container fail to start after the worker node upgrade to SuSE SLES SP4

### ISSUE THREAD

**Issue by calvinxu** [ceph/ceph-csi#4138]
### Describe the bug
csi-rbdplugin container fail to start after the host upgrade to SuSE SLES SP4, as the module files delivered in SP4 with .ko.zst compress format

### Environment details
Image/version of Ceph CSI driver : 3.7.2
Helm chart version : 3.11.1
Kernel version : 5.3.18-57-default
Mounter used for mounting PVC: rbd
Kubernetes cluster version :2.6.6
Ceph cluster version : Quincy(17.2.6) 

### Steps to reproduce
worker node upgrade from SuSE SLES SP3 to SuSE SLES SP4

Setup details: '...'
After OS upgraded to SuSE SLES SP4, the csi-rbdplugin pod( csi-rbdplugin container failed to start)

### Describe what happened
csi-rbdplugin pod started and run normally

### Logs
run a debug container of csi-rbdplugin, check the modprobe version
#modprobe --version
kmod version 25
+XZ +ZLIB +OPENSSL -EXPERIMENTAL

#modprobe rbd -vv
modprobe: INFO: custom logging function 0x55ef36a0cf70 registered
insmod /lib/modules/5.14.21-150400.24.81-default/kernel/net/ceph/libceph.ko.zst
modprobe: INFO: Failed to insert module '/lib/modules/5.14.21-150400.24.81-default/kernel/net/ceph/libceph.ko.zst': Exec format error
modprobe: ERROR: could not insert 'rbd': Exec format error
modprobe: INFO: context 0x55ef388d7600 released

checked that module files delivered in SP4 with .ko.zst compress format 
#ls -trl /lib/modules/`uname -r`/kernel/drivers/block/
total 316
-rw-r--r-- 1 root root 30322 Aug  9 15:59 xen-blkfront.ko.zst
-rw-r--r-- 1 root root 14491 Aug  9 15:59 virtio_blk.ko.zst
-rw-r--r-- 1 root root 13731 Aug  9 15:59 sx8.ko.zst
-rw-r--r-- 1 root root 84135 Aug  9 15:59 rbd.ko.zst
-rw-r--r-- 1 root root 31930 Aug  9 15:59 pktcdvd.ko.zst
-rw-r--r-- 1 root root 32301 Aug  9 15:59 nbd.ko.zst
-rw-r--r-- 1 root root 21557 Aug  9 15:59 loop.ko.zst
-rw-r--r-- 1 root root 52604 Aug  9 15:59 floppy.ko.zst
-rw-r--r-- 1 root root  7556 Aug  9 15:59 brd.ko.zst

similar issue https://github.com/ceph/ceph-csi/issues/3251 and for this, but just wanna to know if any plan CentOS Stream 9( https://github.com/ceph/ceph-container/pull/2038) to be released and Ceph base image to use it?

**Comment by Madhu-1** [comment]
@calvinxu Once we have the required change in the ceph image, we will pull it for cephcsi

**Comment by calvinxu** [comment]
> @calvinxu Once we have the required change in the ceph image, we will pull it for cephcsi

Can I understand that we will get the fix in cephcsi in a future cephcsi release until code change support CentOS Stream 9 merged into ceph-container repo, which is included in the ceph base image?

**Comment by Madhu-1** [comment]
> > @calvinxu Once we have the required change in the ceph image, we will pull it for cephcsi
> 
> Can I understand that we will get the fix in cephcsi in a future cephcsi release until code change support CentOS Stream 9 merged into ceph-container repo, which is included in the ceph base image?

Yes, that's correct, we need to get the ceph base image that supports modprobe of the newer kernel version.

**Comment by calvinxu** [comment]
> > > @calvinxu Once we have the required change in the ceph image, we will pull it for cephcsi
> > 
> > 
> > Can I understand that we will get the fix in cephcsi in a future cephcsi release until code change support CentOS Stream 9 merged into ceph-container repo, which is included in the ceph base image?
> 
> Yes, that's correct, we need to get the ceph base image that supports modprobe of the newer kernel version.
May I know if any target release already planned with this fix? thanks.

**Comment by Madhu-1** [comment]
@calvinxu that you need to check here https://github.com/ceph/ceph-container, once its available we will make use of it in next cephcsi release.

**Comment by nixpanic** [comment]
Depends-on: ceph/ceph-container#2038
